core/templates/dev/head/services/UserService.js

```diff
@@ -19,16 +19,31 @@
 oppia.factory('UserService', [
   '$http', '$q', 'UrlInterpolationService', 'DEFAULT_PROFILE_IMAGE_PATH',
   function($http, $q, UrlInterpolationService, DEFAULT_PROFILE_IMAGE_PATH) {
-    var _isLoggedIn = GLOBALS.userIsLoggedIn;
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
+    var getUserInfoAsync = function() {
+      if (GLOBALS.userIsLoggedIn) {
+        if (this.userInfo) {
+          return $q.resolve(this.userInfo);
+        }
+        return $http.get(
+          '/userinfohandler'
+        ).then(function(response) {
+          this.userInfo = response.data;
```

Comment:
Not sure if you need "that = this". Might also be better to use a local var (like we do elsewhere).

```
    var getUserInfoAsync = function() {
      if (GLOBALS.userIsLoggedIn) {
        if (this.userInfo) {
          return $q.resolve(this.userInfo);
        }

        var that = this;
        return $http.get(
          '/userinfohandler'
        ).then(function(response) {
          that.userInfo = response.data;
          return response.data;
        });
      } else {
        return $q.resolve({});
      }
    };
```